### DISCUSSION TITLE
No `eth_getTransactionByBlockNumberAndIndex` is sent to my RPC

### DISCUSSION THREAD

**Discussion by megtog** [ConsenSys-archive/ganache#3734] [General]
I'm playing with this script (`src/chains/ethereum/ethereum/tests/forking/myscript.ts`). My plan is to understand how mainnet fork works so I just put my script in the `src` directory.

There are `eth_blockNumber` and some `eth_getTransactionCount` requests.  But why is there no `eth_getTransactionByBlockNumberAndIndex`?

 Not sure what I'm doing wrong.

```typescript
import {performance} from 'perf_hooks';
import getProvider from "../helpers/getProvider";
import { EthereumProvider } from "../../src/provider";

async function main() {
  const URL = 'https://localhost:xxxx'; // my RPC endpoint
  console.log(performance.now());
  const blockNumber = 15501711;
  const provider: EthereumProvider = await getProvider({
    fork: {
      url: URL,
      blockNumber,
    }
  });
  const result = await provider.send(
    "eth_getTransactionByBlockNumberAndIndex",
    [`0x${(15501710).toString(16)}`, `0x${(345).toString(16)}`]);
  console.log(performance.now());
  console.log(result);
}

main().then(
  () => { console.log('done'); },
  (err) => { console.log(err); },
);
```

**Comment by megtog** [comment]
It seems that ganache uses `eth_getBlockByNumber(15501710, true)` to get all 348 txs.

Does ganache have local cache to store these data?  It seems the `eth_getBlockByNumber` request would be sent again.

**Reply by davidmurdoch** [reply]
It caches the data. See `src/chains/ethereum/ethereum/src/forking/persistent-cache/persistent-cache.ts` (a disk backed cache) and the `requestCache` (an in-memory cache) in `src/chains/ethereum/ethereum/src/forking/handlers/base-handler.ts`.

To disable the caches set the `fork.disableCache` option to `true` (`getProvider({fork:{disableCache: true}})`).
